### PR TITLE
Deploy Memcached instance in kuttl tests

### PR DESCRIPTION
**Pull request by kajinamit**
We are adding support for caching using memcached globally in all OpenStack services. This is the prep work for it and ensures the CR instance is created in kuttl tests.

This also refactors the entrypoints for all kuttl tests to make all implementations more consistent and simple.

### CODE REVIEW

**Comment by kajinamit** [comment]
recheck

**Comment by kajinamit** [comment]
recheck

**Comment by kajinamit** [comment]
The `openstack_crds` target is removed from kuttl targets.

The `openstack_crds` target depends on `openstack_deploy_prep`, which calls `openstack_deploy_cleanup` and eventually `openstack_cleanup` and `openstack_cleanup` fails unless OpenStackControlPlane CR is created. Because of these dependencies kuttl targets do not work unless openstack-operator is installed, which seems wrong to me.

**Review by bshephar — APPROVED** [review]
Looks good to me. Good amount of planning and thought put into this change tkajinam++

File: Makefile [hunk 1]
```diff
@@ -1256,33 +1255,34 @@ glance_kuttl_run: ## runs kuttl tests for the glance operator, assumes that ever
 	kubectl-kuttl test --config ${GLANCE_KUTTL_CONF} ${GLANCE_KUTTL_DIR}
 
 .PHONY: glance_kuttl
-glance_kuttl: input openstack_crds deploy_cleanup mariadb mariadb_deploy keystone keystone_deploy glance_deploy_prep glance ## runs kuttl tests for the glance operator. Installs openstack and openstack-storage crds, mariadb, keystone and glance operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
+glance_kuttl: kuttl_common_prep glance glance_deploy_prep ## runs kuttl tests for the glance operator. Installs glance operator and cleans up previous deployments before running the tests, add cleanup after running the tests.
```
Comment (kajinamit) [inline Makefile:1258]: glance does not require rabbitmq but I keep it deployed to keep the logic simple. It also allows us to enabe notifications in the future.

File: Makefile [hunk 2]
```diff
@@ -1172,14 +1176,13 @@ ironic_kuttl_run: ## runs kuttl tests for the ironic operator, assumes that ever
 	kubectl-kuttl test --config ${IRONIC_KUTTL_CONF} ${IRONIC_KUTTL_DIR}
 
 .PHONY: ironic_kuttl
-ironic_kuttl: input openstack_crds deploy_cleanup mariadb mariadb_deploy keystone keystone_deploy ironic ironic_deploy_prep ironic_deploy  ## runs kuttl tests for the ironic operator. Installs openstack crds and keystone operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
+ironic_kuttl: kuttl_common_prep ironic ironic_deploy_prep  ## runs kuttl tests for the ironic operator. Installs ironic operator and cleans up previous deployments before running the tests, add cleanup after running the tests.
```
Comment (kajinamit) [inline Makefile:1179]: Current ironic does not require rabbitmq but I keep it deployed so that we can add rabbitmq support in the future.

**Comment by abays** [comment]
recheck

File: Makefile [hunk 3]
```diff
@@ -1147,72 +1159,62 @@ keystone_kuttl_run: ## runs kuttl tests for the keystone operator, assumes that
 keystone_kuttl: export NAMESPACE = ${KEYSTONE_KUTTL_NAMESPACE}
 # Set the value of $KEYSTONE_KUTTL_NAMESPACE if you want to run the keystone
 # kuttl tests in a namespace different than the default (keystone-kuttl-tests)
-keystone_kuttl: input openstack_crds deploy_cleanup mariadb mariadb_deploy mariadb_deploy_validate keystone_deploy_prep keystone ## runs kuttl tests for the keystone operator. Installs openstack crds and keystone operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
+keystone_kuttl: kuttl_db_prep keystone keystone_deploy_prep ## runs kuttl tests for the keystone operator. Installs keystone operator and cleans up previous deployments before running the tests, add cleanup after running the tests.
+	make wait
```
Comment (abays) [inline Makefile:1163]: To use `make wait`, `$(eval $(call vars,$@,keystone))` is needed here.  This is also true for all the other new additions of `make wait` that have been added in this PR (except, of course, `s/keystone/<operator_name>`).  Without it this error occurs:

```
Please set OPERATOR_NAME
+ exit 1
make[1]: *** [Makefile:364: wait] Error 1
```

Comment (kajinamit) [reply Makefile:1164]: Oh, Thanks for pointing this. Fixed it in the latest version.

**Comment by kajinamit** [comment]
recheck

**Comment by abays** [comment]
recheck

**Comment by abays** [comment]
recheck

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**